# DATA PREPROCESSING

### Dataset: Sephora Products and Skincare Reviews

The dataset contains information about all beauty products (over 8,000) from the Sephora online store, including product and brand names, prices, ingredients, ratings, and all features.

In [11]:
import kagglehub

path = kagglehub.dataset_download("nadyinky/sephora-products-and-skincare-reviews")

print("Path to dataset files:", path)

Path to dataset files: C:\Users\tanja\.cache\kagglehub\datasets\nadyinky\sephora-products-and-skincare-reviews\versions\2


In [12]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import warnings
import os


In [13]:
# List all files in the downloaded dataset directory
files = os.listdir(path)
print("Downloaded files:", files)

Downloaded files: ['product_info.csv', 'reviews_0-250.csv', 'reviews_1250-end.csv', 'reviews_250-500.csv', 'reviews_500-750.csv', 'reviews_750-1250.csv']


In [14]:
csv_file = os.path.join(path, "product_info.csv")
df = pd.read_csv(csv_file)

# Check the first few rows of the dataframe
print(df.head())

  product_id               product_name  brand_id brand_name  loves_count  \
0    P473671    Fragrance Discovery Set      6342      19-69         6320   
1    P473668    La Habana Eau de Parfum      6342      19-69         3827   
2    P473662  Rainbow Bar Eau de Parfum      6342      19-69         3253   
3    P473660       Kasbah Eau de Parfum      6342      19-69         3018   
4    P473658  Purple Haze Eau de Parfum      6342      19-69         2691   

   rating  reviews            size                      variation_type  \
0  3.6364     11.0             NaN                                 NaN   
1  4.1538     13.0  3.4 oz/ 100 mL  Size + Concentration + Formulation   
2  4.2500     16.0  3.4 oz/ 100 mL  Size + Concentration + Formulation   
3  4.4762     21.0  3.4 oz/ 100 mL  Size + Concentration + Formulation   
4  3.2308     13.0  3.4 oz/ 100 mL  Size + Concentration + Formulation   

  variation_value  ... online_only out_of_stock  sephora_exclusive  \
0             NaN  ...

In [15]:
print("Shape of the DataFrame:", df.shape)
print(df.info())



Shape of the DataFrame: (8494, 27)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8494 entries, 0 to 8493
Data columns (total 27 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   product_id          8494 non-null   object 
 1   product_name        8494 non-null   object 
 2   brand_id            8494 non-null   int64  
 3   brand_name          8494 non-null   object 
 4   loves_count         8494 non-null   int64  
 5   rating              8216 non-null   float64
 6   reviews             8216 non-null   float64
 7   size                6863 non-null   object 
 8   variation_type      7050 non-null   object 
 9   variation_value     6896 non-null   object 
 10  variation_desc      1250 non-null   object 
 11  ingredients         7549 non-null   object 
 12  price_usd           8494 non-null   float64
 13  value_price_usd     451 non-null    float64
 14  sale_price_usd      270 non-null    float64
 15  limited_edition     

We have 8494 records and 27 features, since we will not use all of them we can identify columns like brand_id and variation_desc, which are redundant or irrelevant for content-based filtering and drop them.

In [18]:
columns_to_drop = [
    'brand_id',
    'variation_type',          # Redundant with brand_name
    'variation_desc',   
    'value_price_usd', 
    'variation_value',  
    'sale_price_usd',   
    'limited_edition',   
    'new',               
    'online_only',       
    'sephora_exclusive', 
    'highlights',  
    'secondary_category',
    'primary_category',   
    'tertiary_category', 
    'child_count',      
    'child_max_price',   
    'child_min_price'    
]

# Drop the columns from the DataFrame:
df_reduced = df.drop(columns=columns_to_drop, axis=1)
print(df_reduced.info())


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8494 entries, 0 to 8493
Data columns (total 10 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   product_id    8494 non-null   object 
 1   product_name  8494 non-null   object 
 2   brand_name    8494 non-null   object 
 3   loves_count   8494 non-null   int64  
 4   rating        8216 non-null   float64
 5   reviews       8216 non-null   float64
 6   size          6863 non-null   object 
 7   ingredients   7549 non-null   object 
 8   price_usd     8494 non-null   float64
 9   out_of_stock  8494 non-null   int64  
dtypes: float64(3), int64(2), object(5)
memory usage: 663.7+ KB
None
